frontend/public/reducers/features.ts

```diff
@@ -58,6 +58,10 @@ export const featureReducer = (state: FeatureState, action: FeatureAction): Feat
   }
 
   switch (action.type) {
+    case ActionType.AddFlag:
+      // eslint-disable-next-line no-console
+      console.log(`${action.payload.flag} is added`);
+      return state.set(action.payload.flag, false);
```

Comment:
The initial value should be set to `undefined` (feature detection in progress).

`false` means the feature detection is complete and feature is not available.

But I don't think we really need another action, like `ActionType.AddFlag`.

The simplest way to handle flags outside the base `FLAGS` enum is to remove unknown flag check for `ActionType.SetFlag`:

```ts
case ActionType.SetFlag:
  return state.set(action.payload.flag, action.payload.value);
```

But I think the best solution is to iterate over all extensions which deal with flags and build an array of allowed flag names, and check against that.